tests/functional/nccl_connection.c

```diff
@@ -27,6 +27,14 @@ int main(int argc, char* argv[])
 
 	MPI_Init(&argc, &argv);
 	MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+	int size;
+	MPI_Comm_size(MPI_COMM_WORLD, &size);
+	if (size != 2) {
+		NCCL_OFI_WARN("Expected two ranks but got %d. "
+			"The nccl_connection functional test should be run with exactly two ranks.",
+			size);
+		return -1;
```

Comment:
Don't return `-1` from main.  Return codes in Unix are interpreted as an unsigned integer and you can get some fun bugs with people trying to interpret `-1`.  Just return `1` or similar positive numbers.